### ISSUE TITLE
Question on ACSRendererView on iOS14 SwiftUI

### ISSUE THREAD

**Issue by hyounoo** [Azure/Communication#171]
I cannot display renderview.createView() UIView on SwiftUI

below function is called when start video call but I cannot get this localRenderView displayed on ContentView

```
    func startLocalVideo() {
        if AVCaptureDevice.authorizationStatus(for: .video) == .authorized {
            let firstCamera: VideoDeviceInfo? = self.deviceManager?.getCameraList()![0]
            let localVideoStream = LocalVideoStream(camera: firstCamera)
            do {
                self.localRenderer = try Renderer(localVideoStream: localVideoStream!)
                self.localRendererView = try self.localRenderer!.createView()
            } catch {
                print(error.localizedDescription)
            }

        } else {
            self.requestDevicePermission()
        }
    }
```

Tried to convert RendererView to UIViewRepresentable, but had no luck.
```
struct RendererView: UIViewRepresentable {

    func makeUIView(context: Context) -> RendererView {
        return RendererView()
    }

    func updateUIView(_ uiView: RendererView, context: Context) {
    }
}
```
![image](https://user-images.githubusercontent.com/9585318/106654236-3799c680-65db-11eb-9467-5419b496dd75.png)

Will this work? or should I go back to iOS13?

**Comment by sankum-msft** [comment]
Hi @hyounoo ! 
There are couple of issues with your code:
 1. You have created a new type with `RendererView` name that is conflicting with the type from ACS SDK. 
 2. `makeUIView` should return the value of `localRendererView` in place of creating a new instance.

Here is the fixed version that should work for you:

```swift
// No changes to your startLocalVideo() method

// Fix#1: Renamed to LocalVideoView
struct LocalVideoView: UIViewRepresentable {

    // This should be passed when you instantiate this class.
    // You should use value of `self.localRendererView` variable in your case.
    let view:RendererView

    func makeUIView(context: Context) -> UIView {
        return view // Fix#2: do NOT create a new view but rather use value returned from `renderer.createView()`
    }

    func updateUIView(_ uiView: UIView, context: Context) {}
}
```

Please let us know if that doesn't work for you for some reason.

**Comment by hyounoo** [comment]
I found the fix..
always find out right after I post the question :(
```
struct VideoRendererView: UIViewRepresentable {
    var renderer: Renderer
    func makeUIView(context: Context) -> RendererView {
        let renderView = try! renderer.createView()
        return renderView
    }

    func updateUIView(_ uiView: RendererView, context: Context) {
    }
}
```

passed in ACSRenderer into UIViewRepresentable and it works well on SwiftUI

**Comment by hyounoo** [comment]
> Hi @hyounoo !
> There are couple of issues with your code:
> 
> 1. You have created a new type with `RendererView` name that is conflicting with the type from ACS SDK.
> 2. `makeUIView` should return the value of `localRendererView` in place of creating a new instance.
> 
> Here is the fixed version that should work for you:
> 
> ```swift
> // No changes to your startLocalVideo() method
> 
> // Fix#1: Renamed to LocalVideoView
> struct LocalVideoView: UIViewRepresentable {
> 
>     // This should be passed when you instantiate this class.
>     // You should use value of `self.localRendererView` variable in your case.
>     let view:RendererView
> 
>     func makeUIView(context: Context) -> UIView {
>         return view // Fix#2: do NOT create a new view but rather use value returned from `renderer.createView()`
>     }
> 
>     func updateUIView(_ uiView: UIView, context: Context) {}
> }
> ```
> 
> Please let us know if that doesn't work for you for some reason.

Oh found out after I post workaround.

Your code looks better!!!
Thank you!
